### PR TITLE
Create release is coming

### PR DESCRIPTION
**Pull request by johnsimons**
Lots of interesting changes in this PR.

I have added Jest and Prettier

This also adds a new workflow, which is suppose to be run all the time for testing.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: Test-Please
+on: push
+env:
+  SA_PASSWORD: 'yourStrong45@Password'
```
Comment (johnsimons) [inline .github/workflows/test.yml:4]: These are just used for the local containers, so not really secrets as such

**Comment by johnsimons** [comment]
I'll be adding more tests for release creation

**Comment by johnsimons** [comment]
@jbristowe are you happy with this PR to be merged?
As I said, I am adding more tests in subsequent ones.

**Review by domenicsim1 — APPROVED** [review]
😎 Looks good! 👍

File: src/adapters/axiosAdapter.ts [hunk 2]
```diff
@@ -13,12 +13,14 @@ export class AxiosAdapter<TResource> implements Adapter<TResource> {
...
-                    "X-Octopus-ApiKey": options.configuration.apiKey
+                    "X-Octopus-ApiKey": options.configuration.apiKey ?? ''
                 },
                 responseType: "json", 
             };
             if (typeof XMLHttpRequest === 'undefined') {
-                config.headers["User-Agent"] = "ts-octopusdeploy";
+                if (config.headers) {
+                    config.headers["User-Agent"] = "ts-octopusdeploy";
+                }
```
Comment (jbristowe) [inline src/adapters/axiosAdapter.ts:23]: We need to incorporate a version number into this string for telemetry. Ideally, the string would look like `<client>/<version> (os-version)`.

Comment (johnsimons) [reply src/adapters/axiosAdapter.ts:23]: Created an issue for it @jbristowe 
https://github.com/OctopusDeploy/api-client.ts/issues/26

**Pull request merged**